src/application-delegate.coffee

```diff
@@ -244,6 +244,17 @@ class ApplicationDelegate
     new Disposable ->
       ipcRenderer.removeListener('context-command', outerCallback)
 
+  onSaveWindowState: (callback) ->
```

Comment:
What about renaming this `onSaveWindowStateRequest` to clarify that it's not called as a result of saving state, but as a result of it being requested by the main process. If you read the code that's clear but it would be good to telegraph that explicitly via the name.